src/map/lua/lua_baseentity.cpp

```diff
@@ -1517,6 +1517,36 @@ uint8 CLuaBaseEntity::getCurrentAction()
     return action;
 }
 
+/************************************************************************
+ *  Function: canUseAbilities()
+ *  Purpose : Sees if a mob has perticular effects that stop it from doing thing
+ *  Example : mob:canUseAbilities()
+ *  Notes   : Can't use on NPCs
+ ************************************************************************/
+
+bool CLuaBaseEntity::canUseAbilities()
+{
+    XI_DEBUG_BREAK_IF(m_PBaseEntity->objtype == TYPE_NPC);
+
+    if (m_PBaseEntity == nullptr)
+    {
+        return false;
+    }
+
+    auto* PEntity = static_cast<CBattleEntity*>(m_PBaseEntity);
+    bool actionNotFree = !(m_PBaseEntity->PAI->IsCurrentState<CInactiveState>() || m_PBaseEntity->PAI->IsCurrentState<CRangeState>() || m_PBaseEntity->PAI->IsCurrentState<CAttackState>());
```

Comment:
This is checking that the mob is either: Auto attacking / Ranged Attacking / Idle.

If its those it returns false, if its anything but theses states then it will return true making the below return says that its not available to do things.

Reply:
There are more states than just those, such as the item state. The current implementation is not expandable for future states that may block actions. `CanChangeState()` is generic and queries whether or not any state is allowed to be actionable in it and does not add technical debt.

tl;dr: checking that does the same thing you want, but better

Reply:
Thanks for the explanation~! Fixed